test/javascripts/adapaters/topic-list-test.js.es6

```diff
@@ -0,0 +1,11 @@
+module("adapter:topic-list");
+
+import { finderFor } from 'discourse/adapters/topic-list';
+
+test("finderFor", function() {
+  var mock = sandbox.mock(Discourse);
+  mock.expects("ajax").withArgs("/search.json?q=test%25%25");
```

Comment:
Ahhh I tried out pretender but it presents a different set of problems. I'll try to summarize what I've done so far. :smile_cat: 

### Problem
Malformed URI error raised by the server because we are passing it a URI that it cannot parse. Example: `/search.json?q=test%`

### What we need to test here
We need to add a test to ensure that the URL we passed to the ajax call encodes the `%` character. In other words, I'm only concerned about the parameter that `Discourse.ajax` receives.

### Problem faced when using pretender
We use `route-recognizer` to identify the route to match to in the pretender. However, [`route-recognizer` decodes the path](https://github.com/discourse/discourse/blob/master/vendor/assets/javascripts/route-recognizer.js#L622) which isn't what the actual server would do. Otherwise, I would have tried something like this.

```javascript
var params;
new Pretender(function() {
  this.get('/search.json', function(request) {
    params = request.queryParams();
  });
}); 
equal(params.q, 'test%25')
```